src/Lab.Quiz.DAL/Repositories/Repository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Lab.Quiz.DAL.Interfaces;
+using Lab.Quiz.DAL.Entities;
+using System.Collections.Generic;
+
+namespace Lab.Quiz.DAL.Repositories
+{
+    public class Repository<TEntity> : IRepository<TEntity> where TEntity : class 
+    {
+        DbSet<TEntity> Entities { get; set; }
+        public Repository(DbSet<TEntity> entities)
+        {
+            Entities = entities;
+        }
+
+        public async Task<TEntity> GetByIdAsync(Guid id)
+        {
+            return await Entities.FindAsync(id);
```

Comment:
In such cases like this method we actually can omit await and async. Anyway the code that will use this method will await it. But it's not very big change, so you can leave it as is.

Reply:
Ok, I got it